Zend/zend_stream.c

```diff
@@ -225,22 +236,22 @@ ZEND_API void zend_file_handle_dtor(zend_file_handle *fh) /* {{{ */
 		efree(fh->buf);
 		fh->buf = NULL;
 	}
-	if (fh->free_filename && fh->filename) {
-		efree((char*)fh->filename);
+	if (fh->filename) {
+		zend_string_release(fh->filename);
 		fh->filename = NULL;
 	}
 }
 /* }}} */
 
 /* return int to be compatible with Zend linked list API */
-ZEND_API int zend_compare_file_handles(zend_file_handle *fh1, zend_file_handle *fh2) /* {{{ */
+static int zend_compare_file_handles(zend_file_handle *fh1, zend_file_handle *fh2) /* {{{ */
```

Comment:
```suggestion
static bool zend_compare_file_handles(zend_file_handle *fh1, zend_file_handle *fh2) /* {{{ */
```
As this is internal now it might make sense to also boolify the return type

Reply:
This function is used as a callback in zend_llist_del_element(). We have to keep "int" here.

Reply:
Ah indeed, I missed this, apologies.